### DISCUSSION TITLE
Getting "_tkinter.TclError: bad screen distance" error when adding any widget on a Toplevel window

### DISCUSSION THREAD

**Discussion by TheRealWushuMaster** [TomSchimansky/CustomTkinter#1939] [Q&A]
# TL;DR
Adding any CTk Widget on a CTkToplevel window yields the error message `_tkinter.TclError: bad screen distance`.

## What I'm using
Python=3.11.3
CustomTkinter=5.2.0

## Context
I have a solar system simulator project that can simulate the trajectory of a spaceship. I have created a canvas that shows the celestial objects and everything is working fine.
In order to have a convenient way for the user to setup the simulation and the spaceships, I want to create a popup window, with some appropriate widgets, where those details can be customized.

## Code walkthrough
I have added a button on a corner in the canvas to be used to open the configuration window, using the following code:
```
self.setup_simulation_button = ctk.CTkButton(self.canvas, text="Setup simulation")
self.button_window = self.canvas.create_window(self.canvas.winfo_reqwidth()-10,
                                               7,
                                               anchor=ctk.NE,
                                               window=self.setup_simulation_button)
```
The method is configured later on, and the property is also initialized:
```
widgets.setup_simulation_button.configure(command=self.show_setup_window)
self.setup_simulation_window = None
```
In the last code snippet, the `widgets` object does not have a `self` in front because it's returned by the method to the main `App` object.

The code for the method that creates the window is as follows:
```
def show_setup_window(self):
    if (self.setup_simulation_window is None
        or not self.setup_simulation_window.winfo_exists()):
        self.setup_simulation_window = SetupSimulationWindow()
    self.setup_simulation_window.focus()
```
Finally, here's the code of the class used for the popup window:
```
class SetupSimulationWindow(ctk.CTkToplevel):
    def __init__(self):
        super().__init__()
        self.geometry("400x300")
        self.title("Simulation Setup")
        self.iconbitmap(default=WINDOW_LOGO)
        self.label = ctk.CTkLabel(master=self, text="Testing using another window")
        self.label.pack(padx=20, pady=20)
```
The class code is only meant to show the window and a couple of widgets for testing purposes, but I haven't been able to make it work. Running the code as is yields the following error message:

> _tkinter.TclError: bad screen distance "0.0"

However, showing the window without any widgets works fine, and the blank window is displayed correctly.

I've noticed that the number provided in the error message, in the last case the "0.0", seems to be related to the width of the widget trying to be shown, but I haven't been able to pinpoint why.
For example, using the following code:

`self.label = ctk.CTkLabel(master=self, text="Testing using another window", width=42)`

yields the following error message:

> _tkinter.TclError: bad screen distance "42.0"

Additionally, I've seen that the same kind of error arises regardless of the widget being created (buttons, labels, frames, etc). The same **does not** apply if I create a regular tkinter widget on the new customtkinter window; somehow, mixing the two packages works fine.

## Last thoughts
There has to be some conflict with something else in my project, because I've tried the example code on the CTkToplevel documentation and it works fine; the window is created and the label is shown. However, I have no idea what to look for.

Please help!

**Comment by Heribert17** [comment]
Do you set the the locale to a country that has "," as separator in float values (for example German) with a call to locale like: locale.setlocale(locale.LC_ALL, "")?
There is a bug in tkinter (since 2011) that some calls to tk convert float values with the locale settings. But TCL isn't able to understand "42,0". It needs "42.0". Deleting the locale call fixes this problem but now you have to format numbers in your code.

**Reply by TheRealWushuMaster** [reply]
Thanks for your answer!

I don't have any locale call set, but I tried changing the system's locale settings to use the dot symbol as the float separator (previously I was using the comma) and now the error has disappeared!

Should I set up the locale to always use the dot as the float separator, regardless of the system's configuration?

**Reply by TheRealWushuMaster** [reply]
I've tried a couple of solutions and found that I can temporarily change the locale settings within the new window to use the dot as the float separator and, afterwards, revert the settings to their original state.

1. Saving the settings: `original_locale = locale.getlocale(locale.LC_NUMERIC)`
2. Changing: `locale.setlocale(locale.LC_NUMERIC, "C")`
3. Reverting: `locale.setlocale(locale.LC_NUMERIC, original_locale)`

This is working so far, so it looks like it might solve the problem.
